src/filter.rs

```diff
@@ -209,47 +207,90 @@ fn filter_internal(
     bpp: usize,
     len: usize,
     previous: &[u8],
-    current: &mut [u8],
+    current: &[u8],
+    output: &mut [u8],
 ) -> FilterType {
     use self::FilterType::*;
 
+    const CHUNK_SIZE: usize = 32;
```

Comment:
Is there any particular reason for this size? Experimental data? So that it fits in a specific SIMD register length for a single-load access?

Reply:
This is based on benchmarking a couple possible sizes. There wasn't a ton of difference between 16, 32, and 64 byte sizes but this option was fastest. The fact that the best size is equal to two 256-bit SIMD registers is probably not a coincidence, however.

Reply:
That might make a good comment. Relying on auto-vectorization in combination with experimental constants it can quickly become hard to distinguish between incidental and intentional constants.

Reply:
That would make a good comment. Relying on auto-vectorization in combination with experimental constants it can quickly become hard to distinguish between incidental and intentional constants.